pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -713,6 +733,10 @@ func (cb *ClusterBuilder) setUpstreamProtocol(cluster *clusterWrapper, port *mod
 	effectiveProxyConfig := cb.proxyMetadata.ProxyConfigOrDefault(cb.req.Push.Mesh.GetDefaultConfig())
 	preserveHeaderCase := effectiveProxyConfig.GetProxyHeaders().GetPreserveHttp1HeaderCase().GetValue()
 
+	if isExplicitHTTP {
```

Comment:
Not super familiar with when the envoyfilter gets applied, but we should probably merge `setHTTP1Options` to also take into account `PreserveHeaderCase` as in the lines below, otherwise values might get unexpectedly overwritten